### PR TITLE
Upgrade bazel to 0.29.1

### PR DESCRIPTION
**Pull request by jtattermusch**
Bring back https://github.com/grpc/grpc/pull/20165 (which has been previously reverted), but this time also upgrade to just-released 0.29.1 (which fixes the bazel bug we hit with 0.29)

### CODE REVIEW

File: tools/remote_build/windows.bazelrc [hunk 1]
```diff
@@ -38,7 +37,7 @@ test --test_env=GRPC_VERBOSITY=debug
 
 # Set flags for uploading to BES in order to view results in the Bazel Build
 # Results UI.
-build --bes_backend="buildeventservice.googleapis.com"
+build --bes_backend=grpcs://buildeventservice.googleapis.com
```
Comment (nicolasnoble) [inline tools/remote_build/windows.bazelrc]: image

**Comment by jtattermusch** [comment]
The win RBE problems:
https://source.cloud.google.com/results/invocations/c3a8a25d-96ca-4c7e-b3b5-48e1eac22828/log
```
T:\src\github\grpc>bazel --bazelrc=tools/remote_build/windows.bazelrc build --invocation_id="33a02e8a-c4a5-428d-a74f-1af0531b2c16" --workspace_status_command=tools/remote_build/workspace_status_kokoro.sh :all --google_credentials=T:\src\gfile/rbe-windows-credentials.json
WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
t:\src\github\grpc/tools/bazel.rc
Starting local Bazel server and connecting to it...
ERROR: Failed to connect to 'grpcs://remotebuildexecution.googleapis.com': cannot find a NameResolver for grpcs://remotebuildexecution.googleapis.com
ERROR: Failed to connect to 'grpcs://remotebuildexecution.googleapis.com': cannot find a NameResolver for grpcs://remotebuildexecution.googleapis.com
INFO: Streaming Build Event Protocol to https://source.cloud.google.com/results/invocations/33a02e8a-c4a5-428d-a74f-1af0531b2c16
```

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/20095

File: tools/remote_build/windows.bazelrc (hunk 1 above)
Comment (gnossen) [reply tools/remote_build/windows.bazelrc]: I love this reaction image, but I have no idea how to interpret it. :joy:

**Pull request merged**